common/document_parser/lib/section_parse/parsers/navy_parser.py

```diff
@@ -0,0 +1,165 @@
+from re import search, IGNORECASE, VERBOSE, compile, finditer, RegexFlag
+from os.path import splitext
+from typing import List
+from .parser_definition import ParserDefinition
+from .utils import (
+    match_number_hyphenated_section,
+    get_letter_dot_section,
+    match_number_dot_section,
+    match_first_appendix_title,
+    remove_pagebreaks,
+    DD_MONTHNAME_YYYY,
+)
+
+
+class NavyParser(ParserDefinition):
+    """Section parser for Navy document types (see SUPPORTED_DOC_TYPES).
+
+    Child of ParserDefinition.
+    """
+
+    SUPPORTED_DOC_TYPES = [
+        "opnavinst",
+        "opnavnote",
+        "bumedinst",
+        "milpersman",
+        "respersman",
+        "bupersinst",
+        "secnavinst",
+        "secnavnote",
+        "mco",
+        "navmc",
+        "comnavresforcominst",
+        "comnavresforcomnote",
+        "bumednote",
+        "alnav",
+        "mcrp",
+        "mcbul",
+        "fmfrp",
+    ]
+
+    def __init__(self, doc_dict: dict, test_mode: bool = False):
+        super().__init__(doc_dict, test_mode)
+        self._text = self.get_raw_text()
+        self._filename_without_extension = splitext(self._filename)[0]
+
```

Comment:
![Screen Shot 2022-11-18 at 11 20 07 AM](https://user-images.githubusercontent.com/37678069/202753423-858cf4b5-8546-41fa-a1bc-1fee592290ca.png)
Is `all_sections` supposed to be a superset of the rest of the sections?

Reply:
@mishoe It's meant to be the entire document text split into sections. When I initially created the DoD parser, we wanted to parse all sections of a document. But, moving forward, we're targeting specific sections like purpose, responsibilities, etc. so `all_sections` isn't populated for parsers created later.